rpc/src/cluster_tpu_info.rs

```diff
@@ -75,8 +75,8 @@ impl TpuInfo for ClusterTpuInfo {
     ) -> Vec<(&SocketAddr, Slot)> {
         let recorder = self.poh_recorder.read().unwrap();
         let leaders: Vec<_> = (0..max_count)
+            .rev()
```

Comment:
what is the purpose of `rev` here? All these leaders get mixed up below in a hashmap right?

Reply:
Also why do we even dedup the leaders here?

Reply:
> Also why do we even dedup the leaders here?

to avoid sending txes to the same leader twice. The caller `SendTransactionService` literally just loops over the returned `Vec` here. so, dedup is rather required as this is done by the counterpart (`get_leader_tpus()`)

Reply:
> what is the purpose of `rev` here? All these leaders get mixed up below in a hashmap right?

hehe. don't blame me. but `.rev()` is needed to return earlier slot to the current slot.
removing that `.rev()` would result in this assertion error in this newly added test:

```
test cluster_tpu_info::test::test_get_leader_tpus ... thread 'cluster_tpu_info::test::test_get_leader_tpus' panicked at rpc/src/cluster_tpu_info.rs:254:9:
assertion `left == right` failed
  left: [(127.0.0.1:2222, 0), (127.0.0.1:1111, 8)]
 right: [(127.0.0.1:2222, 0), (127.0.0.1:1111, 4)]

```

Reply:
Oh lol it's because the hashmap collection will overwrite entries, so we reverse so early slots get inserted **after**.